### PR TITLE
add a python example

### PR DESCRIPTION
**Pull request by jettero**
feel free to move around into a subdir or whatever.

I wanted to use this right away, and I didn't want to use it in javascript, so ... this is a verbatim copy of your javascript with a few python tweaks.

I intend to make a proper python module later, but perhaps someone will beat me to it.

### CODE REVIEW

**Comment by jettero** [comment]
Oh, feel free to decline too. I could just maintain my python port as a forked repo.

**Comment by kexinrong** [comment]
@jettero Thanks for the pull request! Having a python implementation is great =) I will take a look today.

**Comment by kexinrong** [comment]
Sorry I forgot to publish my comment. Otherwise LGTM.

File: ASAP.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        if metrics.kurtosis >= orig_kurt:
+            if metrics.roughness < min_obj:
+                window_size = w
+                min_obj = metrics.roughness
+            head = w + 1
+        else:
+            tail = w - 1
+    return window_size
+
+def SMA(data, _range, slide):
```
Comment (jettero) [inline ASAP.py]: I'm ok with this for sure.

I hadn't put much thought into optimizations at all. I was just trying get it ported more-or-less as-is.

I'll update the PR today.

**Comment by jettero** [comment]
still planning to work on this... delays relating to work

File: ASAP.py (hunk 1 above)
Comment (jettero) [inline ASAP.py]: I finally got around to trying this …

I'm going to have to admit that I don't really understand what SMA() is doing, particularly wrt to the slide. I've done plenty of moving averages ([to name a few](https://github.com/jettero/stockmonkey)). The `moving_average_slide` you're doing here doesn't seem to work out though.

I'll have to put more time into it to figure out why.

Comment (kexinrong) [inline ASAP.py]: moving_average_slide(data, w, s) is calculating the simple moving average of the input time series _data_ with a window size of _w_, and a slide size of _s_. This means that there are _w_ elements in each window, and _s_ elements between neighboring windows. In the following picture for example, we have window size = 5, and slide size = 2.

sma

Hope that the clarification helps!

File: ASAP.py [hunk 2]
```diff
@@ -75,6 +75,18 @@ def SMA(data, _range, slide):
...
+def moving_average(data, _range):
+    ret = numpy.cumsum(data, dtype=float)
+    ret[_range:] = ret[_range:] - ret[:-_range]
+    return ret[_range - 1:] / _range
+
+def moving_average_slide(data, _range, slide):
+    return moving_average(data, _range)[::slide]
+
+x = [42,75,3,5,99,22,88]
+assert SMA(x,3,1) == list(moving_average_slide(x,3,1))
```
Comment (kexinrong) [inline ASAP.py]: FYI I tried this example and I think it failed because SMA() returned wrong values.

**Comment by kexinrong** [comment]
I am going to merge the PR for now since I don't want you to waste more time on SMA implementation. I will take care of it myself. Again, thank you so much for your contribution!

**Comment by jettero** [comment]
I completely lost track of this. I was describing this project to a co-worker and re-found it on github. Thanks for finishing and for the merge.

**Pull request merged**